spec/models/book_spec.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe Book, type: :model do
+  it { should validate_presence_of(:genre) }
+  it { should validate_presence_of(:author) }
+  it { should validate_presence_of(:image) }
+  it { should validate_presence_of(:title) }
+  it { should validate_presence_of(:publisher) }
+  it { should validate_presence_of(:year) }
+
+  subject(:book) { create(:book) }
+
+  it { is_expected.to be_valid }
+
+  describe '#generate_verification_code' do
```

Comment:
which behavior do you want to test here? the book model doesn't generate codes

Reply:
I used the example of card and forgot to eliminate what I would not use